### ISSUE TITLE
kube-linter custom template for "anti-affinity" is not validating the yamls

### ISSUE THREAD

**Issue by kanavdevops** [stackrox/kube-linter#163]
**System info:**
 - OS: [Linux]

**Kube-linter version**
0.2.0
**Describe the bug**
kube-linter custom template for "anti-affinity" is not validating the yamls.
In below kube-linter custom config file i.e config.yaml, minReplicas: 4 and topologyKey: topology.kubernetes.io/zone are configured.
While the test statefulset yaml has 2 replicas and different topologyKey i.e topologyKey: "kubernetes.io/hostname" configured.

The idea behind the check configured is that no one should be able to checkin a k8s yaml with a less of no. of minimum replicas.

The expected behaviour is kube-linter will validate the statefulset yaml  based on the no. replicas and topology key passed from the config yaml, and throw the error that only 2 replicas are configured, also a different topology key is passed.

But kube-linter does not throw any error while performing kube-linter line --config config.yaml .

Statefulset yaml used for testing
```
apiVersion: apps/v1beta1
kind: StatefulSet
metadata:
  labels:
    app: cool-app
  name: cool-app
  namespace: default
spec:
  replicas: 2
  selector:
    matchLabels:
      app: cool-app
  serviceName: cool-app
  template:
    metadata:
      labels:
        app: cool-app
    spec:
      affinity:
        podAntiAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
          - labelSelector:
              matchExpressions:
                - key: "app"
                  operator: In
                  values:
                  - cool-app
            topologyKey: "kubernetes.io/hostname"
      containers:
      - command:
        - sh
        - -c
        - executablefile --process_number_that_uses_statefulset_suffix $(expr $(hostname | grep -o "[[:digit:]]*$"))
        image: us.gcr.io/host/cool-app
        name: cool-app
        resources:
          limits:
            cpu: "6"
            memory: 30Gi
          requests:
            cpu: "6"
            memory: 30Gi
```
Custom kub-linter config file : config.yaml
```
customChecks:
- name: "required-label-app"
  template: "required-label"
  params:
    key: "app"
    value: "cool-app"
- name: "min-replicas-check"
  template: "anti-affinity"
  params:
    minReplicas: 4
    topologyKey: topology.kubernetes.io/zone
```
**To Reproduce**
kube-linter lint --config config.yaml .

**Output**
```
src/kubelinter/test-manifests/test-affinity.yaml: (object: default/cool-app apps/v1beta1, Kind=StatefulSet) container "cool-app" does not have a read-only root file system (check: no-read-only-root-fs, remediation: Set readOnlyRootFilesystem to true in your container's securityContext.)

src/kubelinter/test-manifests/test-affinity.yaml: (object: default/cool-app apps/v1beta1, Kind=StatefulSet) container "cool-app" is not set to runAsNonRoot (check: run-as-non-root, remediation: Set runAsUser to a non-zero number, and runAsNonRoot to true, in your pod or container securityContext. See https://kubernetes.io/docs/tasks/configure-pod-container/security-context/ for more details.)

Error: found 2 lint errors
One or more kubernetes yamls failed kube-linter validation!
```

It does not throw the Error for customCheck "min-replicas-check". Please have a look and suggest

**Comment by viswajithiii** [comment]
Hi @kanavdevops, the anti-affinity check does not validate that every object has the minimum number of replicas -- its purpose is actually to ensure that, in every object that has at least `minReplicas` replicas, the `podAntiAffinity` is defined appropriately. The idea behind this was that users may not want to force configuration of anti-affinity for deployments with a small number of replicas (say, 1).

If I understand correctly, are you saying that you want to check that every object has at least a certain number of replicas? That's not currently supported, but is definitely a feature we can add, and would be a new template. It is not something we expected to need, because it seems unrealistic to require _every_ object to be replicated -- is this your use case? Curious to learn more.

Thanks for filing this issue!

**Comment by kanavdevops** [comment]
thanks for looking, 
However i see from the document https://docs.kubelinter.io/#/generated/templates, we can enforce the no. of minimum replicas that has to be present in the manifests. 

```
 "description": "The minimum number of replicas a deployment must have before anti-affinity is enforced on it",
```
Is there any other way we can configure this template

**Comment by viswajithiii** [comment]
Yeah, I realize that wording is confusing, but what it means is "we don't care about anti-affinity _unless_ a deployment has at least these many replicas".

Can you explain your use-case to me? Do you want _all_ deployments to have anti-affinity, _and_ all deployments to have at least 4 replicas?

**Comment by janisz** [comment]
Closing due to no response
